### PR TITLE
Send of entangled W state implementation

### PR DESCRIPTION
**Pull request by AleMuzzi**
I have implemented the send of an entangled W state (https://en.wikipedia.org/wiki/W_state).
It has been developed analyzing the Qiskit code provided at [this link](https://notebook.community/antoniomezzacapo/qiskit-tutorial/community/teach_me_qiskit_2018/w_state/W%20State%201%20-%20Multi-Qubit%20Systems) generalizing it to a dynamic n-qubit W state generator and writing it accordingly to the already existing GHZ and EPR state exchange.
I have already discussed it a bit with Stephen DiAdamo when I asked him via mail for some informations.
I have already tested the protocol comparing the system density matrix with the ones I found [here](https://arxiv.org/pdf/1807.05572.pdf) at page 6.

I look forward to hearing from you soon,
Alessandro Muzzi

### CODE REVIEW

**Comment by AleMuzzi** [comment]
I added the fidelity commit by error, I would have done a separate pull request but I don't know how to remove the last commit from this request.

**Comment by AleMuzzi** [comment]
The fidelity implementation follows the implementation given [here](https://mathematica.stackexchange.com/a/229556) for Wolfram Mathematica

**Comment by stephendiadamo** [comment]
Thanks for the PR. I'll review it this week.

File: qunetsim/components/host.py [hunk 1]
```diff
@@ -924,6 +925,83 @@ def send_ghz(self, receiver_list, q_id=None, await_ack=False, no_ack=False, dist
...
+                               distribute one
+        Returns:
+            (str, bool): Qubit ID of the shared W and ACK status
+        """
+
+        q_list: list[Qubit] = []
+        n = len(receiver_list) + (0 if distribute else 1)
+
+        def f_gate(i, j, k):
+            theta = math.acos(math.sqrt(1/(n-k+1)))
```
Comment (stephendiadamo) [inline qunetsim/components/host.py]: We're using a single space between mathematical operations: eg `a+b` -> `a + b`

File: requirements.txt [hunk 2]
```diff
@@ -6,3 +6,4 @@ cqc==3.2.2
 simulaqron==3.0.15
 eqsn==0.0.8
 qutip==4.5.2
+scipy~=1.5.4
```
Comment (stephendiadamo) [inline requirements.txt]: Let's use a hard requirement `scipy==1.5.4`

File: qunetsim/objects/qubit.py [hunk 3]
```diff
@@ -113,6 +115,30 @@ def send_to(self, receiver_id):
         """
         self._host.backend.send_qubit_to(self, self._host.host_id, receiver_id)
 
+    def fidelity(self, other_qubit):
```
Comment (stephendiadamo) [inline qunetsim/objects/qubit.py:118]: Can we add a docstring here?

**Review by stephendiadamo — CHANGES_REQUESTED** [review]
Very nice! Just some small changes.
Would you also be interested in generating an example for the online docs?

File: qunetsim/components/host.py (hunk 1 above)
Comment (AleMuzzi) [reply qunetsim/components/host.py]: ok, I'll update that

File: requirements.txt (hunk 2 above)
Comment (AleMuzzi) [reply requirements.txt]: got it

File: qunetsim/objects/qubit.py (hunk 3 above)
Comment (AleMuzzi) [reply qunetsim/objects/qubit.py:118]: yes, of course, I'll add some docs

**Comment by AleMuzzi** [comment]
Of course, how can I do that?

**Comment by stephendiadamo** [comment]
> Of course, how can I do that?

We're using Sphinx to generate documentation, you can see some samples in the `docs` folder .
 I'm going to clone your fork in a second and I can set up a template for you, I'll just need a few minutes and I'll make a push on your branch when it's ready

**Comment by stephendiadamo** [comment]
On this branch I added a template. 
`https://github.com/tqsd/QuNetSim/tree/AleBestia/master`
It's in `docs/source/examples/send_w.rst`. It uses RST style. Once you're done I'll rebuild the docs and push everything on master.

**Comment by AleMuzzi** [comment]
Ok, I'll commit all the changes in these days

**Pull request merged**